src/components/Register.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+export default function Register() {
+  interface User {
+    username: string;
+    password: string;
+    passwordConfirm: string;
+    email: string;
+  }
+  const [newUser, setNewUser] = useState<User>({
+    username: "",
+    password: "",
+    passwordConfirm: "",
+    email: "",
+  });
+  const [errorMsg, setErrorMsg] = useState("");
+  const handleNewUser = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewUser({
+      ...newUser,
+      [e.target.name]: e.target.value,
+    });
+  };
+  const handleSubmit = () => {
+    if (newUser.password !== newUser.passwordConfirm) {
```

Comment:
--- note (no action needed) ---
backend validation will also be implemented (we need to align on what values are legal)

Reply:
noted